solidity/test/InterchainAccountRouter.t.sol

```diff
@@ -129,4 +129,30 @@ contract InterchainAccountRouterTest is Test {
         );
         assertEq(localIca.owner(), address(originRouter));
     }
+
+    function testReceiveValue(uint256 value) public {
+        vm.assume(value <= address(this).balance);
+
+        // receive value before deployed
+        assert(ica.code.length == 0);
+        ica.transfer(value / 2);
+
+        // receive value after deployed
+        remoteRouter.getDeployedInterchainAccount(originDomain, address(this));
+        assert(ica.code.length > 0);
+        ica.transfer(value / 2);
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    function receiveValue() external payable {}
+
+    function testSendValue(uint256 value) public {
```

Comment:
This was a bit confusing too read. IIUC, the ICA makes the call on `this`, and you test that value can be moved. IMO, shoulnd't the test transfer funds to the ICA, and then show that the ICA call can actually be done with value?

Reply:
your suggestion is what the test is doing
1. ica.transfer(value)
2. dispatch this.receiveValue ICA call that transfers value